packages/fxa-payments-server/src/App.tsx

```diff
@@ -50,39 +53,77 @@ export const App = ({
-                exact
-                render={props => (
-                  <SettingsLayout>
-                    <Subscriptions {...props} />
-                  </SettingsLayout>
-                )}
-              />
-              <Route
-                path="/products/:productId"
-                render={props => (
-                  <SignInLayout>
-                    <Product {...props} />
-                  </SignInLayout>
-                )}
-              />
-            </React.Suspense>
-          </Router>
-        </AppContext.Provider>
-      </ReduxProvider>
-    </StripeProvider>
+    <AppContext.Provider value={appContextValue}>
+      <AppErrorBoundary>
+        <StripeProvider apiKey={config.stripe.apiKey}>
+          <ReduxProvider store={store}>
+            <Router>
+              <React.Suspense fallback={<RouteFallback />}>
+                {/* Note: every Route below should also be listed in INDEX_ROUTES in server/lib/server.js */}
+                <Route
+                  path="/"
+                  exact
+                  render={() => <Redirect to="/subscriptions" />}
+                />
+                <Route
+                  path="/subscriptions"
+                  exact
+                  render={props => (
+                    <SettingsLayout>
+                      <Subscriptions {...props} />
+                    </SettingsLayout>
+                  )}
+                />
+                <Route
+                  path="/products/:productId"
+                  render={props => (
+                    <SignInLayout>
+                      <Product {...props} />
+                    </SignInLayout>
+                  )}
+                />
+              </React.Suspense>
+            </Router>
+          </ReduxProvider>
+        </StripeProvider>
+      </AppErrorBoundary>
+    </AppContext.Provider>
+  );
+};
+
+export class AppErrorBoundary extends React.Component {
+  state: {
+    error: undefined | Error;
+  };
+  constructor(props: {}) {
+    super(props);
+    this.state = { error: undefined };
+  }
+  static getDerivedStateFromError(error: Error) {
+    return { error };
+  }
+  componentDidCatch(error: Error, errorInfo: any) {
+    console.error('AppError', error);
+    Sentry.captureException(error);
+  }
+  render() {
+    const { error } = this.state;
+    return error ? <AppErrorDialog error={error} /> : this.props.children;
+  }
+}
+
+export const AppErrorDialog = ({ error: { message } }: { error: Error }) => {
+  const { locationReload } = useContext(AppContext);
+  // TODO: Not displaying the actual error message to the user, just logging it.
+  // Most of these errors will probably be failure to load Stripe widgets.
+  const displayMessage = getErrorMessage('api_connection_error');
+  return (
+    <SettingsLayout>
+      <DialogMessage className="dialog-error" onDismiss={locationReload}>
+        <h4 data-testid="error-loading-app">General application error</h4>
```

Comment:
Not sure about the wording of this heading.  Personally I prefer just "Error".  🤷‍♂